src/domain/program.ts

```diff
@@ -108,8 +109,16 @@ export type ProgramCost = ExecutableCost
 export type ParsedCodeType = {
   encoding: Encoding
   entrypoint: string
-  file: any
-}
+} & (
+  | {
+      file?: undefined
```

Comment:
What is an optional `undefined` type?

Reply:
thats the way typescript have to define "onOf" types in union derived types